infrastructure/src/main/java/org/corfudb/infrastructure/ManagementAgent.java

```diff
@@ -385,36 +385,43 @@ private boolean runRecoveryReconfiguration() {
-        if (layout == null) {
-            return;
-        }
 
-        // Build and replace existing locally stored nodeMetrics
-        SequencerMetrics sequencerMetrics;
-        // This is an optimization. If this node is not the primary sequencer for the current
-        // layout, there is no reason to request metrics from this sequencer.
-        if (layout.getSequencers().get(0).equals(getLocalEndpoint())) {
-            try {
-                sequencerMetrics = CFUtils.getUninterruptibly(
-                        getCorfuRuntime()
-                                .getLayoutView().getRuntimeLayout(layout)
-                                .getSequencerClient(getLocalEndpoint())
-                                .requestMetrics());
-            } catch (ServerNotReadyException snre) {
-                sequencerMetrics = new SequencerMetrics(SequencerStatus.NOT_READY);
-            } catch (Exception e) {
-                log.error("Error while requesting metrics from the sequencer: ", e);
+        try {
+            // Initializing the status of components
+            // No need to poll unless node is bootstrapped.
+            Layout layout = serverContext.copyManagementLayout();
+            if (layout == null) {
+                return;
+            }
+
+            // Build and replace existing locally stored nodeMetrics
+            SequencerMetrics sequencerMetrics;
+            // This is an optimization. If this node is not the primary sequencer for the current
+            // layout, there is no reason to request metrics from this sequencer.
+            if (layout.getSequencers().get(0).equals(getLocalEndpoint())) {
+                try {
+                    sequencerMetrics = CFUtils.getUninterruptibly(
+                            getCorfuRuntime()
+                                    .getLayoutView().getRuntimeLayout(layout)
+                                    .getSequencerClient(getLocalEndpoint())
+                                    .requestMetrics());
+                } catch (ServerNotReadyException snre) {
+                    sequencerMetrics = new SequencerMetrics(SequencerStatus.NOT_READY);
+                } catch (Exception e) {
+                    log.error("Error while requesting metrics from the sequencer: ", e);
+                    sequencerMetrics = new SequencerMetrics(SequencerStatus.UNKNOWN);
+                }
+            } else {
                 sequencerMetrics = new SequencerMetrics(SequencerStatus.UNKNOWN);
             }
-        } else {
-            sequencerMetrics = new SequencerMetrics(SequencerStatus.UNKNOWN);
-        }
 
-        localServerMetrics =
-                new ServerMetrics(NodeLocator.parseString(getLocalEndpoint()), sequencerMetrics);
+            localServerMetrics =
+                    new ServerMetrics(NodeLocator.parseString(getLocalEndpoint()),
+                            sequencerMetrics);
+
+        } catch (Exception e) {
```

Comment:
should this be a method wrapper utility so that you do not have to do it in multiple places.

Reply:
Done.